### PR TITLE
fix(halo2): add engineevm module skeleton

### PR DESCRIPTION
**Pull request by corverroos**
Implements the basic skeleton (including all the protobufs) for the new `engineevm` module

task: none

### CODE REVIEW

File: engineevm/x/engineevm/types/tx.pb.go
Comment (ttarsi) [inline engineevm/x/engineevm/types/tx.pb.go:1]: do we have to keep this in the root dir like this?

**Review by ttarsi — APPROVED** [review]
🙌

File: engineevm/x/engineevm/types/tx.pb.go
Comment (corverroos) [reply engineevm/x/engineevm/types/tx.pb.go:1]: not sure what you mean, the `tx.pb.go` types are usually in the `/x/moduel/types` package, which in this case is `evmengine/types`

Comment (corverroos) [reply engineevm/x/engineevm/types/tx.pb.go:1]: oh sorry yeah, rouge proto file, will delete

**Pull request merged**